### DISCUSSION TITLE
Multiple Parallel Transactions with AsyncLocalStorage

### DISCUSSION THREAD

**Discussion by Mr0nline** [Automattic/mongoose#14895] [Q&A] [answered]
Setting `AsyncLocalStorage` to `true` will help me to avoid passing `session` for all operations within transactions.
My question over here is what if I have multiple transactions running in parallel like
```javascript
// Global
mongoose.set('transactionAsyncLocalStorage', true);

// Service 1
userTransaction = mongoose.connection.transaction("..transaction code")

// Service 2
otherTransaction = mongoose.connection.transaction("...transaction code")
```

Would mongoose will automatically make sure that sessions would be passed but it wouldn't collide with other transactions (i.e. if first transaction fails, it wouldn't roll back second transaction)

**Comment by Mr0nline** [comment]
bumping up!

**Reply by vkarpov15** [reply, marked as answer]
I think this is more of a MongoDB server question, but if the first transaction fails then the 2nd transaction shouldn't get rolled back unless it conflicts with another transaction. Mongoose's `transaction()` function starts a new session every time. Does that help?

**Reply by Mr0nline** [reply]
Thank you! That was what I was trying to find out whether transaction method will create a new session or not!
